packages/nx-plugin/src/utils/testing-utils/nx-project.ts

```diff
@@ -15,7 +15,7 @@ function runNxNewCommand(args?: string, silent?: boolean) {
   return execSync(
     `node ${require.resolve(
       '@nrwl/tao'
-    )} new proj --nx-workspace-root=${localTmpDir} --no-interactive --skip-install --collection=@nrwl/workspace --npmScope=proj --preset=empty ${
+    )} new proj --nx-workspace-root=${localTmpDir} --no-interactive --skip-install --collection=@nrwl/workspace --npmScope=proj --packageManager=yarn --preset=empty ${
```

Comment:
This is a hack and we need conditional logic ... I need some help with this.

Reply:
@Phillip9587 any chance you could give this a review, and lend some help? thx

Reply:
If you need to detect which package manager invoked the process you should use `process.env.npm_config_user_agent`.
```sh
yarn/3.1.1 npm/? node/16.13.0 linux x64
npm/7.20.1 node/v16.13.0 linux x64 workspaces/false
pnpm/6.11.0 npm/? node/v16.13.0 linux x64
```

Reply:
Excellent! I could implement that.

I'm also trying to get it to work using devkit functions .. here is what I've got so far but I am reading documentation as I am unfamiliar with the APIs.
````typescript
# adding host:tree is a breaking change?
function runNxNewCommand(host: Tree, args?: string, silent?: boolean) {
  const localTmpDir = dirname(tmpProjPath());
  const packageManager = detectPackageManager(getWorkspacePath(host));
  
  // getWorkspacePath(host)) 
  console.log('Your package manager is:', detectPackageManager());

    return execSync(
      `node ${require.resolve(
        '@nrwl/tao'
      )} new proj --nx-workspace-root=${localTmpDir} --no-interactive --skip-install --collection=@nrwl/workspace --npmScope=proj --packageManager=${packageManager} --preset=empty ${
        args || ''
      }`,
      {
        cwd: localTmpDir,
        ...(silent && false ? { stdio: ['ignore', 'ignore', 'ignore'] } : {}),
      }
    );
  
}
````

Reply:
For anyone following along here... as of `Nx 13.3.x` ... without any changes to `@nrwl/nx-plugin` I get both a `package-lock.json` and `yarn.lock` in the e2e folder
![image](https://user-images.githubusercontent.com/707567/143785472-d4fa489b-ce73-484b-acbe-091827bab28b.png)

We're trying to modify this `runNxNewCommand()` function to prevent that

Reply:
Given
![image](https://user-images.githubusercontent.com/707567/143786531-72455913-f804-4f2c-87d6-2d5b1d72f5c7.png)

Fail when executing `nx run e2e-domain-cypress-domain-test:e2e --verbose` as `process.env.npm_config_user_agent` doesn't pick up on `npm` when `nx` is invoked directly
![image](https://user-images.githubusercontent.com/707567/143786167-7655ddd2-2c2c-4044-af65-f28cf72c178b.png)

Success when executing: `yarn nx run e2e-domain-cypress-domain-test:e2e --verbose`
![image](https://user-images.githubusercontent.com/707567/143786177-7c5a2c67-dfeb-4866-b8f3-06f2ccb936a3.png)

Success when executing: `yarn dlx nx run e2e-domain-cypress-domain-test:e2e --verbose`
![image](https://user-images.githubusercontent.com/707567/143786204-c9a2441f-ec09-4d91-b6e7-754de900a938.png)

Surprisingly... SUCCESS when executing: `nx run e2e-domain-cypress-domain-test:e2e --verbose` and `npx nx run e2e-domain-cypress-domain-test:e2e --verbose`
![image](https://user-images.githubusercontent.com/707567/143786236-ddafc0e3-fafb-498e-be7a-44d15289df4b.png)

But then I figured out that it was just a false-positive because of Nx Cache. This fails: `nx run e2e-domain-cypress-domain-test:e2e --verbose --skip-nx-cache`
![image](https://user-images.githubusercontent.com/707567/143786367-ec59dee0-5de3-49ea-82ef-f3c771a12f73.png)

However, note that this passed `npx nx run e2e-domain-cypress-domain-test:e2e --verbose --skip-nx-cache`
![image](https://user-images.githubusercontent.com/707567/143786432-0789eebe-3268-407f-9189-54e364ad63d7.png)

Reply:
Screenshots showing the appearance of `package-lock.json` and 'yarn.lock` given different commands

![image](https://user-images.githubusercontent.com/707567/143786681-c4cb1d91-1dae-4546-b364-192157849fb4.png)

![image](https://user-images.githubusercontent.com/707567/143786833-db6dd9ae-d0ac-4eed-8e40-00ef5a9bc4f9.png)

And note that `pnpx` fails.
![image](https://user-images.githubusercontent.com/707567/143787025-00ddca69-31ef-4322-933e-2ebf3aaa27b1.png)

I am going to push my code as-is so others can jump in on this.

Reply:
I think this is a better approach for `packages/nx-plugin/src/utils/testing-utils/nx-project.ts` ... I coped code from `packages/create-nx-plugin`

````typescript
import { appRootPath } from '@nrwl/tao/src/utils/app-root';
import {
  getPackageManagerCommand,
  PackageManager,
  readJsonFile,
  writeJsonFile,
} from '@nrwl/devkit';
import { execSync } from 'child_process';
import { dirname } from 'path';
import { ensureDirSync } from 'fs-extra';
import { tmpProjPath } from './paths';
import { cleanup } from './utils';
import * as yargsParser from 'yargs-parser';

 // START copy functionality from: '../../../../create-nx-plugin/bin/detect-invoked-package-manager'
 // would like to import, but it needs to be refactored to @nrwl/devkit

 /**
  * Detects which package manager was used to invoke create-nx-{plugin|workspace} command
  * based on the main Module process that invokes the command
  * - npx returns 'npm'
  * - pnpx returns 'pnpm'
  * - yarn create returns 'yarn'
  *
  * Default to 'npm'
  */
 export function detectInvokedPackageManager(): PackageManager {
   let detectedPackageManager: PackageManager = 'yarn'; // changed from 'npm'
   // mainModule is deprecated since Node 14, fallback for older versions
   const invoker = require.main || process['mainModule'];
 
   // default to `yarn`
   if (!invoker) {
     return detectedPackageManager;
   }
 
   for (const pkgManager of ['pnpm', 'yarn', 'npm'] as const) {
     if (invoker.path.includes(pkgManager)) {
       detectedPackageManager = pkgManager;
       break;
     }
   }
 
   return detectedPackageManager;
 }

 // END copy
 //

function runNxNewCommand(args?: string, silent?: boolean) {
  const localTmpDir = dirname(tmpProjPath());

  // copied from: packages/create-nx-plugin/bin/create-nx-plugin.ts
  const parsedArgs: any = yargsParser(process.argv.slice(2), {
    string: [
      'name',
      'cli',
      'preset',
      'appName',
      'style',
      'defaultBase',
      'packageManager',
    ],
    alias: {
      packageManager: 'pm',
    },
    boolean: ['help', 'interactive', 'nxCloud'],
    default: {
      interactive: false,
    },
    configuration: {
      'strip-dashed': true,
      'strip-aliased': true,
    },
  }) as any;

  const packageManager: PackageManager =
  parsedArgs.packageManager || detectInvokedPackageManager();

  // amends the value to an string, even if its undefined or null etc
  console.log(packageManager);

  return execSync(
    `node ${require.resolve(
      '@nrwl/tao'
    )} new proj --nx-workspace-root=${localTmpDir} --packageManager=${packageManager} --no-interactive --skip-install --collection=@nrwl/workspace --npmScope=proj --preset=empty ${
      args || ''
    }`,
    {
      cwd: localTmpDir,
      ...(silent && false ? { stdio: ['ignore', 'ignore', 'ignore'] } : {}),
    }
  );
}

export function patchPackageJsonForPlugin(
  npmPackageName: string,
  distPath: string
) {
  const path = tmpProjPath('package.json');
  const json = readJsonFile(path);
  json.devDependencies[npmPackageName] = `file:${appRootPath}/${distPath}`;
  writeJsonFile(path, json);
}

/**
 * Generate a unique name for running CLI commands
 * @param prefix
 *
 * @returns `'<prefix><random number>'`
 */
export function uniq(prefix: string) {
  return `${prefix}${Math.floor(Math.random() * 10000000)}`;
}

/**
 * Run the appropriate package manager install command in the e2e directory
 * @param silent silent output from the install
 */
export function runPackageManagerInstall(silent: boolean = true) {
  const pmc = getPackageManagerCommand();
  const install = execSync(pmc.install, {
    cwd: tmpProjPath(),
    ...(silent ? { stdio: ['ignore', 'ignore', 'ignore'] } : {}),
  });
  return install ? install.toString() : '';
}

/**
 * Creates a new nx project in the e2e directory
 *
 * @param npmPackageName package name to test
 * @param pluginDistPath dist path where the plugin was outputted to
 */
export function newNxProject(
  npmPackageName: string,
  pluginDistPath: string
): void {
  cleanup();
  runNxNewCommand('', true);
  patchPackageJsonForPlugin(npmPackageName, pluginDistPath);
  runPackageManagerInstall();
}

/**
 * Ensures that a project has been setup in the e2e directory
 * It will also copy `@nrwl` packages to the e2e directory
 */
export function ensureNxProject(
  npmPackageName?: string,
  pluginDistPath?: string
): void {
  ensureDirSync(tmpProjPath());
  newNxProject(npmPackageName, pluginDistPath);
}

````

I don't have it working yet, but I think its the right path forward.

Reply:
@benpsnyder Sorry, I don't have any time at the moment...